tests/integ/modin/series/test_between.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime as dt
+
+import pandas as native_pd
+import pytest
+
+import modin.pandas as pd
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from tests.integ.modin.utils import eval_snowpark_pandas_result, create_test_series
+from tests.integ.utils.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=1)
+def test_series_between_default_inclusive():
+    eval_snowpark_pandas_result(
+        *create_test_series(list(range(0, 10))), lambda ser: ser.between(2, 8)
+    )
+
+
+# tuples of (data, low, high)
+BETWEEN_TEST_ARGUMENTS = [
+    ([0.8, 1.1, 0.9, 1.2], 0.9, 1.1),
+    ([0.8, -1.1, 0.9, 1.2], -1, 1.14),
+    # strings are compared lexicographically
+    (["quick", "brown", "fox", "Quick", "Brown", "Fox"], "Zeta", "alpha"),
+    (["quick", "brown", "fox", "Quick", "Brown", "Fox"], "Delta", "kappa"),
+    (["", "aa", "aaaa"], "", "aaa"),
+    (
+        [
+            dt.datetime(2024, 10, 11, 17, 5),
+            dt.datetime(2020, 1, 2, 2, 40),
+            dt.datetime(1998, 7, 7, 12, 33),
+        ],
+        dt.datetime(2019, 1, 1, 0, 0),
+        dt.datetime(2021, 1, 1, 0, 0),
+    ),
+]
+
+
+@sql_count_checker(query_count=1)
+@pytest.mark.parametrize("data, low, high", BETWEEN_TEST_ARGUMENTS)
+@pytest.mark.parametrize("inclusive", ["both", "neither", "left", "right"])
+def test_series_between(data, low, high, inclusive):
```

Comment:
We should also test when `low` and `high` are series. I think it will just work.

Reply:
added